### ISSUE TITLE
"preg_match(): No ending delimiter '/' found" with multiple array rule "example.*.*" in Request file

### ISSUE THREAD

**Issue by rykon97** [laravel/framework#40971]
- Laravel Version: 9.0.0
- PHP Version: 8.1.2
- Database Driver & Version:

### Description:
I have a rule in my request file:
```php
'timeFrom.*.*' => ['regex:/^(00|01|02|03|04|05|06|06|08|09|10|11|12|13|14|15|16|17|18|19|20|21|22|23)\:[0|3][0]$/i'],
'timeTo.*.*' => [ 'regex:/^(00|01|02|03|04|05|06|06|08|09|10|11|12|13|14|15|16|17|18|19|20|21|22|23|24)\:[0|3][0]$/i'],
```
When validations run I have an error: "preg_match(): No ending delimiter '/' found". I make rule with array pattern, but I have error. I think this throws only with "example.* rules, because when I checked for just "timeTo" rule, I have no error.
(Sorry for my English)

### Steps To Reproduce:
Create ExapleRequest with array of rules:

```php
public function rules(): array
    {
        return [
            'timeFrom.*.*' => ['regex:/^(00|01|02|03|04|05|06|06|08|09|10|11|12|13|14|15|16|17|18|19|20|21|22|23)\:[0|3][0]$/i'],
            'timeTo.*.*' => ['regex:/^(00|01|02|03|04|05|06|06|08|09|10|11|12|13|14|15|16|17|18|19|20|21|22|23|24)\:[0|3][0]$/i'],
        ];
    }
```
Then check it with request. It throws an error "preg_match(): No ending delimiter '/' found".

**Comment by derekmd** [comment]
Related: https://github.com/laravel/framework/issues/40924

PR https://github.com/laravel/framework/pull/40941 will fix this once the v9.0.3 release is tagged.

**Comment by rykon97** [comment]
> Related: #40924
> 
> PR #40941 will fix this once the v9.0.3 release is tagged.

Ok, thanks!

I convert my rule to:
```php
'timeFrom.*.*' => [Rule::forEach(static function () {
    return [
        'regex:/^(0[0-9]|(1[0-9]|2[0-3]))\:(0|3)0$/I'
    ];
})],
```

With 9.0.2 works good!
